.github/workflows/esp32_build.yml

```diff
@@ -1,6 +1,30 @@
 name: ESP32 Build
 
-on: [push, pull_request, workflow_dispatch]
+on:
+  push:
+    paths:
+      - '.github/workflows/esp32_build.yml'
```

Comment:
What's your reasoning behind using a whitelist here?  Feels somewhat wrong... for example you're not forcing a test when stuff is committed into the Rover directory...

Reply:
Yes because we build only copter and plane as part of the test, so testing on Rover change would just be a waste as we don't test the changes

Reply:
> Yes because we build only copter and plane as part of the test, so testing on Rover change would just be a waste as we don't test the changes

Fair enough - bit I  reckon if I had to ask the question it deserves a comment.

I still think this would be better as a blacklist rather than as a whitelist - I'd much rather us do more work than necessary than miss a bug in CI because we forgot to add something to a whitelist.